go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -274,13 +274,15 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 		return addErrors
 	}
 
-	if nsInfo.addressSet == nil {
-		// TODO(trozet): remove dependency on nsInfo object and just determine hash names to create Egress FW with
-		return fmt.Errorf("unable to add egress firewall, namespace: %s has no address set", egressFirewall.Namespace)
+	// EgressFirewall needs to make sure that the address_set for the namespace exists independently of the namespace object
+	// so that OVN doesn't get unresolved references to the address_set.
+	// TODO: This should go away once we do something like refcounting for address_sets.
+	err = oc.addressSetFactory.EnsureAddressSet(egressFirewall.Namespace)
+	if err != nil {
+		return fmt.Errorf("cannot Ensure that addressSet for namespace %s exists %v", egressFirewall.Namespace, err)
 	}
-
-	ipv4HashedAS, ipv6HashedAS := nsInfo.addressSet.GetASHashNames()
-	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt, txn)
+	ipv4HashedAS, ipv6HashedAS := addressset.MakeAddressSetHashNames(egressFirewall.Namespace)
+	err = oc.addEgressFirewallRules(ef, ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt, txn)
```

Comment:
@JacobTanenbaum you don't need to pass egressFirewall.Namespace, isn't it? since you can get that info from `ef`

Reply:
removed the call

Reply:
it looks like its still here in this commit, maybe you fixed it in another commit?

Reply:
it was on the wrong commit, It should be fixed now